### PR TITLE
PBFT bugfixes and algorithm improvements

### PR DESCRIPTION
**Pull request by bridger-herman**
Scope: Commits https://github.com/bitwiseio/sawtooth-pbft/pull/2/commits/dbde9eac40626fc6ef3e2be53654190f19fbe70e through https://github.com/bitwiseio/sawtooth-pbft/pull/2/commits/eb99d6bc41fbb94361ecc6272d19eb828f3a046e (10 commits)

Changes introduced:
+ Removes the unnecessary `NewView` and `CommitFinal` message types
+ Changes the way view changes work 
  + Now the timer is only started on `BlockNew`, rather than always running
  + Invalid `BlockNew` updates and `BlockInvalid` updates trigger a view change
+ Adds the ability to visualize PBFT using Grafana
+ Generally cleans up the project

### CODE REVIEW

File: tests/client.yaml [hunk 1]
```diff
@@ -20,9 +20,11 @@ volumes:
 
 services:
   validator-0:
-    image: sawtooth-validator:$ISOLATION_ID
+    image: sawtooth-pbft-test:$ISOLATION_ID
```
Comment (agunde406) [inline tests/client.yaml:23]: Can you add more information to the commit message to explain why these changes belong in the same commit?

File: src/message_extensions.rs [hunk 2]
```diff
@@ -48,14 +52,28 @@ impl Hash for PbftMessageInfo {
     }
 }
 
+impl Hash for PbftBlock {
+    fn hash<H: Hasher>(&self, state: &mut H) {
```
Comment (agunde406) [inline src/message_extensions.rs]: Does this clean up a clippy lint?

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: src/message_extensions.rs (hunk 2 above)
Comment (bridger-herman) [reply src/message_extensions.rs]: Yes - clippy was upset that I was using the protobuf-derived `PartialEq` but providing my own version of `Hash` (`derive_hash_xor_eq`). I made certain that the property `k1 == k2 ==>  hash(k1) == hash(k2)` holds - see comment at the top of the file.

File: tests/client.yaml (hunk 1 above)
Comment (bridger-herman) [reply tests/client.yaml:23]: Addressed

File: src/message_extensions.rs (hunk 2 above)
Comment (agunde406) [reply src/message_extensions.rs]: This commit should be broken up more for each change that was needed as it is hard to understand with all these changes together

File: src/node.rs [hunk 3]
```diff
@@ -356,6 +326,11 @@ impl PbftNode {
     pub fn on_block_new(&mut self, block: Block) -> Result<(), PbftError> {
         info!("{}: Got BlockNew: {:?}", self.state, block.block_id);
 
+        // TODO: Check consensus seal here. If anything in seal is suspicious, then start a view
```
Comment (jsmitchell) [inline src/node.rs]: It would be better to manage TODOs outside of the source (e.g. Jira)

**Review by jsmitchell — APPROVED** [review]

File: src/node.rs (hunk 3 above)
Comment (bridger-herman) [reply src/node.rs]: Addressed

**Review by agunde406 — APPROVED** [review]
